codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsEc2.java

```diff
@@ -87,10 +87,10 @@ public void generateSharedComponents(GenerationContext context) {
                        + "  output: $T,\n"
                        + "  data: any\n"
                        + "): string => {", "};", responseType, () -> {
-
-            // Attempt to fetch the error code from the specific location, including the wrapper.
-            writer.openBlock("if (data.Errors.Error.Code !== undefined) {", "}", () -> {
-                writer.write("return data.Errors.Error.Code;");
+            // Attempt to fetch the error code from the specific location.
```

Comment:
I think we should take a look at moving as much code as possible into library functions written in TypeScript. These kinds of methods seem like they could be generic methods that are HTTP-specific.